### ISSUE TITLE
Improve handling of types

### ISSUE THREAD

**Issue by decibel** [theory/pgtap#91]
This affects the function tests (IE: function_owner_is), but might impact others.

The current API of passing in a name is problematic because if you pass in a non-canonical type name (ie: int) you end up with a very confusing 'function does not exist' error. It would be better if we were dealing with regtype[].

Any significant downsides to changing the parameters of the function_* test functions? FWIW, regtype[] exists back to at least 9.0.

**Comment by theory** [comment]
No, I think that would be fine, as long as we updated `compat/install-8.4.patch` to switch it back to `NAME[]`.

**Comment by decibel** [comment]
Is it worth it to continue supporting 8.4? it's now 2 versions past EOL.

**Comment by theory** [comment]
pgTAP has patch files for support back to 8.1. I dropped 8.0 back in Nov 2011, so I'd say we could start dropping the other 8.x versions. Hasn't really been a reason to since then, though. If the change is just to the signature (`regtype[]` instead of `name[]`), then it will be easy for me to update the patches.
